### ISSUE TITLE
Devices before iPhone 6 may not work

### ISSUE THREAD

**Issue by Slavaleriya** [nanoscopic/ios_remote_provider#37]
Hi! please tell me if there are any restrictions on the connection of this or that device model
For example iPhone 7 does not work with other devices. When connected, either one iPhone7 works or all the others.
And iPhone5SE does not connect at all

**Comment by nanoscopic** [comment]
I have previously had multiple iPhone 7 working simultaneously with at least one other device type at the same time.

Any device that can run the latest iOS should work. I have yet to run through a full test of all device types though.

Devices before iPhone 6s cannot run the latest iOS, and so will need a specially compiled version of WDA just for them.

I'm also unsure if the current WDA master version from Appium ( which my fork is based on ) will work against iPhone 6 and before.

It is certainly possible to build an older version of WDA that is compatible, and then use that to make it work, but I have not done so. My intent so far is to simply not support those older devices.

I am happy to support them if someone does the work for it though. That work is to go and figure out what version of WDA works and let me know.

There is an additional problem as well; I'm not entirely sure that older devices will work with the way I have written the CFStream app. It uses ReplayKit2. I think ReplayKit1 would be needed for older devices. It is a hassle to rebuild the app using ReplayKit1; so I have no particular interest in doing so. Should one of my paid clients request it I'd put forth the effort for them, but otherwise I have no plan to do so.

At some point I will enable other alternative video options besides CFStream app though, and that would potentially work with the older devices.

TLDR: Yes. Older devices don't work currently.

What exactly is the use case for supporting such old devices? I am more apt to recommend those devices for the trash can.

--

In regard to multiple device types at once; I will be investigating that. I've seen similar people saying there is some issue there and I believe it. Need to replicate the problem and figure out what is going on.

**Comment by Slavaleriya** [comment]
Thanks for the answer!!
iPhone7 is fully functional

But please take a look at 5SE logs, these are the errors that appear. Still, you can't do anything with it?


WARN[0000] Leftover iosif - Sending SIGTERM              pid=38865 proc=iosif type=proc_cleanup
INFO[0001] Process start - device_trigger                binary=bin/iosif proc=device_trigger type=proc_start
Location from redirect of controlfloor login:'https://localhost:8080/provider/'
Logged in to control floor
self signed option
link for dialer = wss://localhost:8080/provider/ws
Creating IIFDev with udid=67c1bd4f66f712e61b676ce532e4b793990e5ae6
udid: 67c1bd4f66f712e61b676ce532e4b793990e5ae6
Running bin/iosif [mg -json -id 67c1bd4f66f712e61b676ce532e4b793990e5ae6 AvailableDisplayZoomSizes main-screen-width main-screen-height ArtworkTraits]
json:{
"AvailableDisplayZoomSizes":{
  default:[
  ]
  zoomed:[
  ]
}
,
"main-screen-width":640
,
"main-screen-height":1136
,
"ArtworkTraits":{
  ArtworkDeviceIdiom:"phone"
  ArtworkDeviceSubType:568
  CompatibleDeviceFallback:"0"
  GraphicsFeatureSetFallbacks:"APPLE3v1:APPLE2:APPLE1:GLES2,0"
  ArtworkDisplayGamut:"sRGB"
  ArtworkDeviceProductDescription:"iPhone SE (1st generation)"
  ArtworkDynamicDisplayMode:"0"
  GraphicsFeatureSetClass:"APPLE3"
  ArtworkDeviceScaleFactor:2
  DevicePerformanceMemoryClass:2
}

}

panic: runtime error: index out of range [1] with length 0

goroutine 9 [running]:
github.com/nanoscopic/ujsonin/v2/mod.(*JArr).GetAt(0xc0002a0120, 0x1, 0x7, 0x447ad20)
 /Users/mcloud-ios/go/pkg/mod/github.com/nanoscopic/ujsonin/v2@v2.0.4/mod/ujsonin.go:77 +0x4b
main.(*DeviceTracker).onDeviceConnect1(0xc00011db90, 0x447b400, 0xc0000b22c0, 0xc0000be340)
 /Users/mcloud-ios/ios_remote_provider/device_tracker.go:128 +0x4a2
main.NewDeviceTracker.func1(0x447b400, 0xc0000b22c0, 0xc00009a060, 0x28)
 /Users/mcloud-ios/ios_remote_provider/device_tracker.go:46 +0x3e
main.(*IIFBridge).OnConnect(0xc00007c680, 0xc00009a060, 0x28, 0xc00009e180, 0x6, 0xc00011dc00)
 /Users/mcloud-ios/ios_remote_provider/bridge_iif.go:88 +0xc8
main.(*IIFBridge).startDetect.func2(0xc0000ca000, 0x4e, 0xc00011dc00)
 /Users/mcloud-ios/ios_remote_provider/bridge_iif.go:64 +0x29a
main.proc_generic.func1(0xc00011dc00, 0xc00010bc50, 0xc00015e100, 0xc00010bb90, 0xc00010bc20, 0xc000100120, 0xc000024c68, 0x0, 0x0)
 /Users/mcloud-ios/ios_remote_provider/proc_generic.go:194 +0xdab
created by main.proc_generic
 /Users/mcloud-ios/ios_remote_provider/proc_generic.go:109 +0x669

**Comment by nanoscopic** [comment]
Have you followed the instructions and added device specific configuration with the width/height of your device? The error code here is indicating that to me. Of course, that is an issue as well, a better error message should be output, but that is what this error is caused by I believe.

**Comment by nanoscopic** [comment]
I don't have any devices before iPhone 6 so unless a paying client wants me to work on this I won't.
